app/controllers/server.rb

```diff
@@ -3,7 +3,45 @@ class Server < Sinatra::Base
     not_found do
       erb :error
     end
-    
-    # Write the rest of your controller code here!
+
+    post '/sources' do
+      Client.validate_and_add_client(params)
```

Comment:
i like your idea of extracting out the logic of your controller - i know sal talked to y'all about where these methods belong. they don't quite belong in `Client` or `Payload` (on line 12). i'd suggest making a module or separate class that handles this logic